aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,150 @@
+
+* **Singleton** shares state across all users of the application and will lead to inappropriate concurrent use.
+* **Scoped** (the default) poses a similar issue between components for the same user.
+* **Transient** results in a new instance per request, but as components can be long-lived this results in a longer-lived context than may be intended.
+
+## Use of DbContext in Blazor Server Apps
+
+The following recommendations are designed to provide a consistent approach to using EF Core in Blazor Server apps. 
+
+* By default, consider using one context per operation. The context is designed for fast, low overhead instantiation:
+
+    ```csharp
+    var using context = new MyContext();
+    return await context.MyEntities.ToListAsync();
+    ```
+* Use a flag to prevent multiple concurrent operations:
+
+    ```csharp
+    if (Loading)
+    {
+        return;
+    }
+    try 
+    {
+        Loading = true;
+        // operations go here
+    }
+    finally 
+    {
+        Loading = false;
+    }
+    ```
+
+* For longer-lived operations that take advantage of EF Core's [change tracking](https://docs.microsoft.com/ef/core/querying/tracking) and/or [concurrency control](https://docs.microsoft.com/ef/core/saving/concurrency), [scope the context to the lifetime of the component](#scope-to-the-component-lifetime).
+
+### Create new DbContext instances
+
+The fastest way to create a new `DbContext` instance is by using `new` to create a new instance. However, there are several scenarios that may require additional dependencies to be resolved. For example, you may wish to use [DbContextOptions](https://docs.microsoft.com/ef/core/miscellaneous/configuring-dbcontext#configuring-dbcontextoptions) to configure the context. 
+
+Here is an example that configures SQLite and enables data logging. The code uses an extension method to configure the database factory for dependency injection and provide the default options:
+
+:::moniker range=">= aspnetcore-5.0"
+[!code-csharp[](./common/samples/5.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Startup.cs?range=29-31)]
+:::moniker-end
+:::moniker range="< aspnetcore-5.0"
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Startup.cs?range=29-31)]
+:::moniker-end
+
+:::moniker range=">= aspnetcore-5.0"
+The recommended solution to create a new `DbContext` with dependencies is to use a factory. EF Core 5.0 provides a built-in factory for creating new contexts.
+:::moniker-end
+:::moniker range="< aspnetcore-5.0"
+The recommended solution to create a new `DbContext` with dependencies is to use a factory. The sample app implements its own factory in `Data/DbContextFactory.cs`. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
+:::moniker-end
+
```

Comment:
Since the factory is introduced in the second paragraph, I think the bit about registering the factory should be moved afterwards.

```suggestion
:::moniker range=">= aspnetcore-5.0"
The recommended solution to create a new `DbContext` with dependencies is to use a factory. EF Core 5.0 provides a built-in factory for creating new contexts.
:::moniker-end
:::moniker range="< aspnetcore-5.0"
The recommended solution to create a new `DbContext` with dependencies is to use a factory. The sample app implements its own factory in `Data/DbContextFactory.cs`. 

[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
:::moniker-end

Here is an example that configures SQLite and enables data logging. The code uses an extension method to configure the database factory for dependency injection and provide the default options:

:::moniker range=">= aspnetcore-5.0"
[!code-csharp[](./common/samples/5.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Startup.cs?range=29-31)]
:::moniker-end
:::moniker range="< aspnetcore-5.0"
[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Startup.cs?range=29-31)]
:::moniker-end

```